astropy/time/core.py

```diff
@@ -1969,7 +1991,7 @@ class TimeISOT(TimeString):
                 '{year:d}-{mon:02d}-{day:02d}'))
 
 
-class TimeYearDayTime(TimeString):
+class TimeYearDayTime(TimeISO):
     """
     Year, day-of-year and time as "YYYY:DOY:HH:MM:SS.sss...".
```

Comment:
I was expecting to see something like `'(?P<year>[+-]\d{5})-%m-%dT%H:%M:%S(?P<scale>(...)?'` here to take advantage of the new regex capability.  Is that not possible or desirable?